### ISSUE TITLE
Same color entries on same column

### ISSUE THREAD

**Issue by tiptobi** [alamkanak/Android-Week-View#131]
Hello,
In my app, I show one day with some events. Some of the events are overlapping but there are no overlapping events with the same color. What I want to reach is, that all events with the same color are drawn in the same "column". 
At the moment it looks like this:
![screenshot_2015-05-22-13-30-52](https://cloud.githubusercontent.com/assets/4949694/7769844/268fa71a-0088-11e5-9cb0-796f9a96f37c.png)

What I'd like to have is, that the blue events are on the first column and the green events on the second. I tried to add all blue events first and then add all green events, but that didn't help. Any ideas, how to solve it?

**Comment by lolobosse** [comment]
Hey tiptobi,

I don't know how much time you want to invest to have this feature (because I don't see any easy way to proceed) but you can change directly the lib at the `private void drawEvents(Calendar date, float startFromPixel, Canvas canvas)` (line 543 in `WeekView`), I don't exactly remember where the events are drawn but I guess that you can play with this (it is, at least, a good startpoint). The problem that I see in your idea is: what happens when one event takes the whole width?

Best,

**Comment by lolobosse** [comment]
@alamkanak @caske33 @subbuboyapati I thought a bit about that issue. Don't you think it could be an enhancement ?

**Comment by caske33** [comment]
Yes, certainly, I have needed this feature too. But at the time, I couldn't find an easy way to fix it (didn't have a lot of time back then)

But I'm guessing, the next pseudo-algorithm might do it:
1. Use the current algorithm to decide how many columns there are at every hour level. (at this moment the events are sorted on time). 
2. Fill the columns when the events are sorted on priority (which would be a new field in the event)

But I'm thinking that the first part might need some alterations too, becasue if the order is changed in the second part, the number of columns might change too.
